fdbrpc/LoadBalance.actor.h

```diff
@@ -202,8 +202,10 @@ Future< REPLY_TYPE(Request) > loadBalance(
 		double nextMetric = 1e9;
 		double bestTime = 1e9;
 		double nextTime = 1e9;
+		int badServers = 0;
+
 		for(int i=0; i<alternatives->size(); i++) {
-			if(bestMetric < 1e8 && i == alternatives->countBest()) {
+			if(badServers < std::min(i, FLOW_KNOBS->LOAD_BALANCE_MAX_BAD_OPTIONS) && i == alternatives->countBest()) {
```

Comment:
The name of this is a little confusing to me -- it's more like the number of bad options for which you would decide to include non-best options. I think of max bad options more like this number minus 1, which is to say the most bad options you can have without triggering the other behavior.

In ratekeeper we have a similar knob `MAX_MACHINES_FALLING_BEHIND` which I think this number is being set to match. I'm not necessarily opposed to having two separate knobs for this if we think there's value in controlling them separately, but if there isn't we could use the same one.

Also, in ratekeeper we have an additional check that it won't let more machines fall behind than `teamSize-1`. I think that's sort of what we're trying to do here, too. By adding that additional constraint, we would make it so that single replicated clusters would also send reads to the other side if the local side is penalized. Setting the max bad options to team size or larger is pretty similar to disabling `LOAD_BALANCE_PENALTY_IS_BAD`, so that may be enough to cover all the cases that we want.

Reply:
I think there might be scenarios where we would want to set this to something different than the ratekeeper knob, s